build.gradle

```diff
@@ -53,6 +53,9 @@ java {
         languageVersion.set(projectJavaVersion)
     }
 }
+runClient {
```

Comment:
If this is really, really neccessary you should not add this to `build.gradle`, but a separate file called `addon.gradle`. It will be used in the build process. And what dependency requires this?

Reply:
Yes, one of the upstream dependences is now causing this error without it when running in dev:

```
java.lang.VerifyError: Expecting a stackmap frame at branch target 11
Exception Details:
  Location:
    net/minecraft/client/renderer/InventoryEffectRenderer.drawScreen(IIF)V @4: ifeq
  Reason:
    Expected stackmap frame at this location.
  Bytecode:
    0x0000000: 2ab4 003d 9900 072a b700 422a 1b1c 25b7
    0x0000010: 0044 b1                                
  Stackmap Table:
    same_frame(@18)

        at net.minecraft.client.Minecraft.runTick(Minecraft.java:1986) ~[Minecraft.class:?]
        at net.minecraft.client.Minecraft.runGameLoop(Minecraft.java:1039) ~[Minecraft.class:?]
        at net.minecraft.client.Minecraft.run(Minecraft.java:962) [Minecraft.class:?]
        at net.minecraft.client.main.Main.main(Main.java:164) [Main.class:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_302]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_302]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_302]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_302]
        at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
        at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
        at net.minecraftforge.gradle.GradleStartCommon.launch(Unknown Source) [start/:?]

```

```
        UCHIJAAAA       mcp{9.05} [Minecraft Coder Pack] (minecraft.jar) 
        UCHIJAAAA       FML{7.10.99.99} [Forge Mod Loader] (forgeSrc-1.7.10-10.13.4.1614-1.7.10.jar) 
        UCHIJAAAA       Forge{10.13.4.1614} [Minecraft Forge] (forgeSrc-1.7.10-10.13.4.1614-1.7.10.jar) 
        UCHIJAAAA       NotEnoughItems{2.1.17-GTNH} [NotEnoughItems] (NotEnoughItems-master-SNAPSHOT-dev.jar) 
        UCHIJAAAA       CodeChickenCore{1.0.9} [CodeChicken Core] (minecraft.jar) 
        UCHIJAAAA       spongemixins{1.3.0} [SpongeMixins Loader] (SpongeMixins-1.3.3-dev.jar) 
        UCHIJAAAA       IC2{2.2.828-experimental} [IndustrialCraft 2] (industrialcraft-2-2.2.828-experimental-dev.jar) 
        UCHIJAAAA       structurelib{1.0.12} [StructureLib] (StructureLib-master-SNAPSHOT-dev.jar) 
        UCHIJAAAA       gregtech{MC1710} [GregTech5-Unofficial] (GT5-Unofficial-master-SNAPSHOT-dev.jar) 
        UCHIJAAAA       hodgepodge{1.6.13} [Hodgepodge] (hodgepodge-1.7.10-1.6.13-3-g543487a.dirty.jar) 

```

Also where is `addon.gradle` referenced? I looked for any other obvious includes in the included script and didn't see it.

Reply:
CRTL-F: `addon.gradle`

https://github.com/GTNewHorizons/Hodgepodge/blob/543487a3dfaed5f274d69e2191bcfd98b222ba49/build.gradle#L190

Reply:
```
compile("com.github.GTNewHorizons:GT5-Unofficial:master-SNAPSHOT:dev") {
    transitive = false
    setChanging(true)
}

// GT5U Dependencies
compile("net.industrial-craft:industrialcraft-2:2.2.828-experimental:dev")
compile("com.github.GTNewHorizons:StructureLib:master-SNAPSHOT:dev")
compile("com.github.GTNewHorizons:NotEnoughItems:master-SNAPSHOT:dev")
```

It must be one of those. `compileOnly` will NOT be loaded at runtime. Most likely, someone "patched" the build script of GT5U or its transitive dependencies (Galacticraft is one, but should work now)

Reply:
hah, apparently I'm blind, thanks

Reply:
I didn't see any obvious changes in CCC/CCL that would introduce this crash.  Will try to track it down because it's annoying me, but this change at least let's it run for now.

Reply:
Based on some limited reading, it seems to be java 7 or earlier bytecode that's (now) invalid... which -noverify ignores....

Reply:
AFAIK InventoryEffectRenderer is only patched by hodgepodge itself. Try change [this line](https://github.com/GTNewHorizons/Hodgepodge/blob/543487a3dfaed5f274d69e2191bcfd98b222ba49/src/main/java/com/mitchej123/hodgepodge/asm/util/AbstractClassTransformer.java#L76) to `new ClassWriter(2)` and see if it works.

Reply:
Will take a look. Thanks for the idea